### PR TITLE
Allow isMapboxHTTPURL to be configured at build time

### PR DESCRIPTION
**Pull request by arindam1993**

### CODE REVIEW

File: src/util/config.js [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 type Config = {|
   API_URL: string,
+  API_URL_REG_EXP: RegExp,
```
Comment (karimnaaji) [inline src/util/config.js:5]: nit: API_URL_REG_EXP -> API_URL_REGEX?

File: src/util/config.js [hunk 2]
```diff
@@ -12,8 +13,22 @@ type Config = {|
   MAX_PARALLEL_IMAGE_REQUESTS: number
 |};
 
+let mapboxHTTPURLRe;
+
 const config: Config = {
     API_URL: 'https://api.mapbox.com',
+    get API_URL_REG_EXP () {
+        if (mapboxHTTPURLRe == null) {
+            const prodMapboxHHTPURLRe = /^((https?:)?\/\/)?([^\/]+\.)?mapbox\.c(n|om)(\/|\?|$)/i;
```
Comment (karimnaaji) [inline src/util/config.js:22]: prodMapboxHHTPURLRe -> prodMapboxHTTPURLRegex?

**Review by karimnaaji — APPROVED** [review]
Looks good, left some minor naming nits.

**Comment by Tofandel** [comment]
Hmm process.env should definitely not be left in a build, when bundling with webpack if the env is not found webpack will just output the whole env as an object instead.. A config option instead of an environnement variable should be used

Or if kept as an env variable then it should at least have a name unique to mapbox

**Comment by Tofandel** [comment]
This is what I mean, this is a production build
image

**Comment by Tofandel** [comment]
Can we bump this as a security issue as this can leak env information without the developers knowledge?

**Comment by stepankuzmin** [comment]
Hi @Tofandel,

Thanks for pointing this out! We'll fix that in https://github.com/mapbox/mapbox-gl-js/issues/12681

**Comment by stepankuzmin** [comment]
Hi @Tofandel,

Unfortunately, I couldn't reproduce the issue with leaking environment variables in the bundle using the latest webpack version (`v5.81.0`, `v5.59.0`). Webpack issues an error if the `API_URL_REGEX` environment variable is missing when using `new EnvironmentPlugin(['API_URL_REGEX'])`:

```
ERROR in EnvironmentPlugin - API_URL_REGEX environment variable is undefined.

You can pass an object with default values to suppress this warning.
See https://webpack.js.org/plugins/environment-plugin for example.
```

If you omit the `API_URL_REGEX` environment variable, webpack won't replace `process.env.API_URL_REGEX`.

We appreciate your concern and understand the problem. While we will not consider this a security issue, we will do our best to address it as soon as possible. Again, we apologize for any inconvenience this may have caused and appreciate your patience. Please let us know if you have any other concerns.

**Comment by Tofandel** [comment]
Hello, In my case it is caused by the [default configuration of `vue-cli` ](https://github.com/vuejs/vue-cli/blob/dev/packages/%40vue/cli-service/lib/config/base.js#L202)

To reproduce it, simply use `webpack.DefinePlugin({ 'process.env': JSON.stringify(process.env) })`

**Pull request merged**